test/Test/Db.hs

```diff
@@ -52,26 +57,34 @@ upsertReposSpec env = describe "upsertRepos" $ do
         env & reposRowDifference (upsertRepos [updatedValidRepo]) `equals` [Id 1 `WithId` updatedValidRepo]
 
 getReposSpec :: AppEnv -> Spec
-getReposSpec env = describe "getRepos" $
-    it "should return a list of repos with length 1" $ do
+getReposSpec env = describe "getRepos" $ do
+    it "should return a list of repos with length 1" $
         env & reposRowCount `equals` 1
-        env & reposUnaffected getRepos `equals` True 
+    it "should should not affect the repos table" $
+        env & reposUnaffected getRepos `equals` True
+
+getReposByCategoriesSpec :: AppEnv -> Spec
+getReposByCategoriesSpec env = describe "getReposByCategories" $ do
+    it "should return a list of repos with length 1" $
+        env & rowsReturned (getReposByCategories [Category "FFI"]) `equals` 1
+    it "should not affect the repos table" $
+        env & reposUnaffected (getReposByCategories []) `equals` True
```

Comment:
I think we don't have to be too paranoid about own code. We use `query` with SELECT there, it won't affect any rows.

Reply:
@chshersh Got it :+1: